projects/smoke_adt7476a/src/main.c

```diff
@@ -0,0 +1,97 @@
+#include "adt7476a_fan_controller.h"
+#include "adt7476a_fan_controller_defs.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+/*
+
+  this teset will increment the fan speed by 10% every second until at full speed,
+  and then loop back to 0% speed and continue.
+
+  the speed interval and increment defines can be changed for further testing
+
+  the i2c port, i2c write and read address, and i2c settings should be changed as needed
+  depending on the board.
+
+*/
```

Comment:
other styling stuff: we only use single line comments, not sure why but at this point it's legacy